lib/utils.js

```diff
@@ -9,8 +9,10 @@ exports.defineCachedRelations = defineCachedRelations;
 exports.sortObjectsByIds = sortObjectsByIds;
 exports.setScopeValuesFromWhere = setScopeValuesFromWhere;
 exports.mergeQuery = mergeQuery;
+exports.createPromiseCallback = createPromiseCallback;
 
 var traverse = require('traverse');
+var Promise = require('bluebird');
```

Comment:
Actually, I don't like the idea of shipping "bluebird" with loopback-datasource-juggler. We should let the user to decide what promise implementation he wants to use and always use `global.Promise`.

There are several options how can `createPromiseCallback` handle the case when `global.Promise` is not defined:
 1. Throw an error explaining the problem. Problem: this is a breaking change, as `User.create(data)` will now throw even though the user does not care about the result of the operation.
 2. Return a no-op callback (same as now) without a promise. Problem: user will get a runtime error while accessing `.then` of undefined, it is totally not clear how to fix the problem.
 3. Return a no-op callback with a dummy promise-like object that throws when `.then` or `.catch` is called.

I am proposing to go with the third option, as it is backward compatible and allows us to print a helpful error message.

A mock implementation:

```js
function createPromiseCallback() {
  var cb;

  if (!global.Promise) {
    cb = function(){};
    cb.promise = {};
    Object.defineProperty(cb.promise, 'then', { get: throwPromiseNotDefined });
    Object.defineProperty(cb.promise, 'catch', { get: throwPromiseNotDefined });
    return cb;
  }

  var promise = new Promise(function (resolve, reject) {
    cb = function (err, data) {
      if (err) return reject(err);
      return resolve(data);
    };
  });
  cb.promise = promise;
  return cb;
}

function throwPromiseNotDefined() {
  throw new Error(
    'Your Node runtime does support ES6 Promises. ' +
    'Set "global.Promise" to your preferred implementation of promises.');
}
```